benchmarks/bm_runner.py

```diff
@@ -537,6 +537,66 @@ def func(args: argparse.Namespace) -> None:
         _subprocess_runner([args.asv_sub_command, *args.asv_args], asv=True)
 
 
+class TrialRun(_SubParserGenerator):
+    name = "trialrun"
+    description = (
+        "Fast trial-run a given benchmark, to check it works : "
+        "in a provided or latest-lockfile environment, "
+        "with no repeats for accuracy of measurement."
+    )
+    epilog = (
+        "e.g. python bm_runner.py trialrun "
+        "MyBenchmarks.time_calc /tmp/testpython/bin/python"
+        "\n  NOTE: setting $DATA_GEN_PYTHON is equivalent to the "
+        "'runpath' argument."
+    )
+
+    def add_arguments(self) -> None:
+        self.subparser.add_argument(
+            "benchmark",
+            type=str,
+            help=(
+                "A benchmark name, possibly including wildcards, "
+                "as supported by the ASV '--benchmark' argument."
+            ),
+        )
+        self.subparser.add_argument(
+            "runpath",
+            type=str,
+            nargs="?",
```

Comment:
Why `nargs` instead of [`required`](https://docs.python.org/3/library/argparse.html#required)?

Reply:
Possibly related - if I don't provide any arguments I get a misleading error message:

```
$ python bm_runner.py trialrun       
usage: bm_runner.py trialrun [-h] benchmark [runpath] ...
bm_runner.py trialrun: error: the following arguments are required: benchmark, asv_args
```

Misleading because `asv_args` is not actually required

Reply:
Is `nargs` perhaps to allow commands like this to work fine (`--version` being an ASV argument)?

```
python bm_runner.py trialrun CubeCreation --version
```

Reply:
> Why nargs instead of [required](https://docs.python.org/3/library/argparse.html#required)?
I initially tried "required=False", but it causes an error - it says it's not allowed for positional arguments.

Reply:
OK I read it up + I think that "nargs='?'" **_is_** correct + intended usage.
https://docs.python.org/3/library/argparse.html#nargs : where it says 
> If no command-line argument is present, the value from [default](https://docs.python.org/3/library/argparse.html#default) will be produced. Note that for optional arguments, there is an additional case ...

I think that implies that it does make sense for positionals too, and the following examples seem to confirm it.

Reply:
> a misleading error message

This appears to be just standard, unfortunately..
```
>>> parser = argparse.ArgumentParser()
>>> _ = parser.add_argument("x")
>>> _ = parser.add_argument("y", nargs="*")
>>> parser.parse_args(["asb", "def"])
Namespace(x='asb', y=['def'])
>>> parser.parse_args(["asb"])
Namespace(x='asb', y=[])
>>> parser.parse_args([])
usage: [-h] x [y ...]
: error: the following arguments are required: x, y

>>>
```

So, the summary shows 'y' as being optional, but the error message doesn't say that :-(

Reply:
.. and here's another oddity.

The "argparse.REMAINDER" constant appears to be undocumented, but it translates to "...".
The effect is odd though, as seen from the error when you don't give a benchmark :

With "nargs=argparse.REMAINDER" :
`usage: bm_runner.py trialrun [-h] benchmark [runpath] ...`
But with "nargs='*'" :
`usage: bm_runner.py trialrun [-h] benchmark [runpath] [asv_args ...]`
(the rest being unchanged)

So, I think the latter is more correct + I've now adopted `nargs="*"`.

Sadly, the syntax error message is still unchanged
`bm_runner.py trialrun: error: the following arguments are required: benchmark, asv_args`
 -- i.e. still "wrong" about the optional nature of asv_args.

On second thoughts though, perhaps an interpretation of that message is that, since it consumes any number of args, including zero, then it _is_ always present (and, coincidentally, always "required").